src/HttpReply.h

```diff
@@ -51,7 +51,8 @@ class HttpReply: public Http::Message
 
     HttpHdrSc *surrogate_control;
 
-    HttpHdrContRange *content_range;
+    /// \returns parsed Content-Range for a 206 response and nil for others
+    HttpHdrContRange *contentRange() const;
```

Comment:
I do not think this PR should add a getter method that duplicates content_range data member functionality. See my [response](https://github.com/squid-cache/squid/pull/77#discussion_r148012095) to your first review rebuttal for more details.

Reply:
This method should be changed into const HttpHdrContRange *contentRange() const;
Then it does not duplicate content_range, and improves the data encapsulation right? I have checked that this works, making const reference argument inside 'HttpHdrRange::contains(HttpHdrRangeSpec& r) const' method (another valuable improvement).

Note that Squid has a huge number of 'public' fields accessed from everywhere and changed from everywhere, and that is a big problem. Improving this (when easy) is the right direction, I assume.

Reply:
> contentRange() should be changed to return `const HttpHdrContRange *`. Then it does not duplicate content_range, and improves the data encapsulation right?

Yes, adding a const-returning method might be viewed as an improvement over direct access to `content_range` but it is a small improvement and feels out of scope here.

One could argue that hiding `content_range` helps prevent similar bugs in other Squid branches (when this fix is ported there), but for that argument to be valid, we would have to hide `HdrType::CONTENT_RANGE` as well!

Overall, I still doubt we should add this method in this project, but if you insist, I can accept it. Your call.

> Squid has a huge number of 'public' fields accessed from everywhere and changed from everywhere, and that is a big problem. Improving this (when easy) is the right direction, I assume.

Protecting fields that should not be accessed directly is indeed a good idea, but mixing such improvements with _unrelated_ bug fixes is not a good strategy.

Reply:
> Your call.

I decided to keep this method, with a minor modification, needed for:

> we would have to hide HdrType::CONTENT_RANGE as well!

I found a solution with the above contentRange(), how to eliminate HdrType::CONTENT_RANGE use from buildRangeHeader(). The only place where it is still used is low-level HttpHeader.cc code.

Reply:
Unfortunately, removing current raw `CONTENT_RANGE` from most of the code is not very useful if new code (or old code in other branches) can access the raw header as needed again. To hide X, we have to make it inaccessible or, at the very least, difficult to access without raising large red flags during review. Not using X is very different from hiding X!

I really do not like the new unparsed API. It needlessly complicates an already tricky API. Please remove it. We could add another dedicated method to access the raw value but, as discussed above, it would not hide that raw value. Truly hiding it requires a new API that works for many other similar cases with cached/parsed header fields. This goes way beyond this issue scope. Let's just access the raw header as needed (and only when needed) instead.

Reply:
Removed.